File: solution/300/solution.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        if(nums.size()<2)return nums.size();
        vector<int> result(nums.size(), 1);
        int max=1;
        for(int i = 1;i < nums.size();i++){
            for(int j=0;j<i;j++){
                if(nums[j]<nums[i] && result[j] + 1 > result[i]){
                    result[i] = result[j] + 1;
                    if(max<result[i])max = result[i];
                }
            }
        }
        return max;
    }
};

File: solution/300/solution.py
class Solution(object):
    def lengthOfLIS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) < 2:return len(nums)
        result = 1
        lenlist = [1] * len(nums)
        for i in xrange(1, len(nums)):
            for j in xrange(i):
                if nums[i] > nums[j] and lenlist[i] < lenlist[j] + 1:
                    lenlist[i] = lenlist[j] + 1
                    if result < lenlist[i]:
                        result = lenlist[i]
        return result